### PR TITLE
Improve metadata handling

### PR DESCRIPTION
**Pull request by trickeydan**
- Default Metadata that can be set.
- Test Coverage 100%
- Handle `JSONDecodeError` nicely

### CODE REVIEW

File: sbot/metadata.py [hunk 1]
```diff
@@ -61,7 +65,10 @@ def _find_file(search_path: str) -> Optional[str]:
 
 def _read_file(path: str) -> Dict[str, Any]:
     with open(path) as file:
-        obj = json.load(file)
+        try:
+            obj = json.load(file)
+        except json.decoder.JSONDecodeError:
+            raise RuntimeError("Unable to decode metadata. Ask a volunteer for help.")
```
Comment (kierdavis) [inline sbot/metadata.py]: I'm not sure I'm a fan of saying things like "Ask a volunteer for help." in error messages. However, I would be in favour of installing a global error handler that prints some brief debugging advice at the end of the traceback, including something along the lines of "if all else fails ask a volunteer for help".

File: tests/test_metadata.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    data = load()
+    assert data == {}
+    data = load(fallback={"beans": 5})
+    assert data == {"beans": 5}
+
+
+def test_load_file() -> None:
+    """Test that we can load a file."""
+    data_path = Path(__file__).parent
+    environ[METADATA_ENV_VAR] = str(data_path.absolute())
```
Comment (kierdavis) [inline tests/test_metadata.py]: Playing with the environment variables in tests is kinda ugly, and it relies on the assumption that this variable does not already exist in the environment - I think `test_load_no_env_var` will fail if it is set. I'm thinking we should decouple the environment variable interaction from the `load` function to make the latter easier to test.

Comment (kierdavis) [reply tests/test_metadata.py]: Tracked in #36.

File: sbot/metadata.py (hunk 1 above)
Comment (trickeydan) [reply sbot/metadata.py]: I deliberately added it here because bad metadata is unlikely to be their fault. 

I'm ok with it in `sbot` only.

**Pull request merged**